### PR TITLE
testing (core): move existing tests to integration tests, rewrite `message.go` unit tests

### PR DESCRIPTION
**Pull request by jimjbrettj**
## Changes

- Move existing core tests to be integration tests
- rewrite message.go unit tests to be faster and increase code coverage

## Tests

```
go test ./dot/core -tags=integration
go test ./dot/core
```

## Issues

-

## Primary Reviewer

- @timwu20

### CODE REVIEW

File: dot/core/messages.go [hunk 1]
```diff
@@ -5,16 +5,53 @@ package core
...
+	ts, err := s.storageState.TrieState(&head.StateRoot)
+	if err != nil {
+		return nil, false, fmt.Errorf("cannot get trie state from storage for root %s: %w", head.StateRoot, err)
+	}
+
+	rt.SetContextStorage(ts)
+
+	// validate each transaction
+	externalExt := types.Extrinsic(append([]byte{byte(types.TxnExternal)}, tx...))
+	val, err := rt.ValidateTransaction(externalExt)
```
Comment (qdm12) [inline dot/core/messages.go:33]: Change this to `validity, err =`

File: dot/core/messages.go [hunk 2]
```diff
@@ -5,16 +5,53 @@ package core
...
+
+	"github.com/libp2p/go-libp2p-core/peer"
 )
 
+func (s *Service) validateTransaction(peerID peer.ID, head *types.Header, rt runtime.Instance,
+	tx types.Extrinsic) (validity *transaction.Validity, valid bool, err error) {
+	s.storageState.Lock()
+	defer s.storageState.Unlock()
+
+	ts, err := s.storageState.TrieState(&head.StateRoot)
```
Comment (qdm12) [inline dot/core/messages.go:24]: nit to unlock it earlier?

```suggestion
	s.storageState.Lock()
	ts, err := s.storageState.TrieState(&head.StateRoot)
	s.storageState.Unlock()
```

File: dot/core/messages.go [hunk 3]
```diff
@@ -41,56 +78,29 @@ func (s *Service) HandleTransactionMessage(peerID peer.ID, msg *network.Transact
...
-			// push to the transaction queue of BABE session
-			hash := s.transactionState.AddToPool(vtx)
-			logger.Tracef("added transaction with hash %s to pool", hash)
+		validity, isValidTxn, err := s.validateTransaction(peerID, head, rt, tx)
+		if err != nil {
+			return false, fmt.Errorf("failed validating transaction for peerID %s: %w", peerID, err)
+		}
 
+		if !isValidTxn {
+			allTxsAreValid = false
```
Comment (qdm12) [inline dot/core/messages.go:89]: should we break here?

File: dot/core/messages_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+	require.NoError(t, err)
+
+	rv, err := rt.Version()
+	require.NoError(t, err)
+
+	c, err := ctypes.NewCall(meta, "System.remark", []byte{0xab, 0xcd})
+	require.NoError(t, err)
+
+	ext := ctypes.NewExtrinsic(c)
+	o := ctypes.SignatureOptions{
```
Comment (qdm12) [inline dot/core/messages_integration_test.go:51]: nit this could had been `options` 🤵

File: dot/core/messages_integration_test.go [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+func TestService_HandleBlockProduced(t *testing.T) {
+	net := new(mocks.Network)
+	cfg := &Config{
+		Network:  net,
+		Keystore: keystore.NewGlobalKeystore(),
+	}
+
+	s := NewTestService(t, cfg)
+	err := s.Start()
+	require.Nil(t, err)
```
Comment (qdm12) [inline dot/core/messages_integration_test.go:81]:
```suggestion
	require.NoError(t, err)
```

**Review by timwu20 — APPROVED** [review]

File: dot/core/messages_integration_test.go [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+	net.On("GossipMessage", expected)
+
+	state, err := s.storageState.TrieState(nil)
+	require.NoError(t, err)
+
+	err = s.HandleBlockProduced(&newBlock, state)
+	require.NoError(t, err)
+
+	time.Sleep(time.Second)
```
Comment (qdm12) [inline dot/core/messages_integration_test.go:116]: Do we need this sleep?

File: dot/core/messages_integration_test.go [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+	require.NoError(t, err)
+	require.True(t, b)
+
+	pending := s.transactionState.(*state.TransactionState).Pending()
+	require.NotEqual(t, 0, len(pending))
+	require.Equal(t, extBytes, pending[0].Extrinsic)
+
+	extBytes = []byte(`bogus extrinsic`)
+	msg = &network.TransactionMessage{Extrinsics: []types.Extrinsic{extBytes}}
+	b, err = s.HandleTransactionMessage(peer1, msg)
```
Comment (qdm12) [inline dot/core/messages_integration_test.go:169]: What's `b` here?

File: dot/core/messages_integration_test.go [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+	require.NoError(t, err)
+
+	extBytes := createExtrinsic(t, rt, genHash, 0)
+	msg := &network.TransactionMessage{Extrinsics: []types.Extrinsic{extBytes}}
+	b, err := s.HandleTransactionMessage(peer1, msg)
+	require.NoError(t, err)
+	require.True(t, b)
+
+	pending := s.transactionState.(*state.TransactionState).Pending()
+	require.NotEqual(t, 0, len(pending))
```
Comment (qdm12) [inline dot/core/messages_integration_test.go:164]:
```suggestion
	require.NotEmpty(t, pending)
```

File: dot/core/messages_integration_test.go [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+	rt.SetContextStorage(ts)
+
+	block := sync.BuildBlock(t, rt, genHeader, nil)
+
+	err = s.handleBlock(block, ts)
+	require.NoError(t, err)
+
+	extBytes := createExtrinsic(t, rt, genHash, 0)
+	msg := &network.TransactionMessage{Extrinsics: []types.Extrinsic{extBytes}}
+	b, err := s.HandleTransactionMessage(peer1, msg)
```
Comment (qdm12) [inline dot/core/messages_integration_test.go:159]: what's `b` here? Maybe a better name?

File: dot/core/service_test.go [hunk 10]
```diff
@@ -2,941 +2,3 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 package core
```
Comment (qdm12) [inline dot/core/service_test.go:4]: delete this file

File: dot/core/messages_integration_test.go (hunk 7 above)
Comment (jimjbrettj) [reply dot/core/messages_integration_test.go:169]: I am not sure, i just moved existing tests over I didn't write the integration tests in this PR

File: dot/core/messages_integration_test.go (hunk 9 above)
Comment (jimjbrettj) [reply dot/core/messages_integration_test.go:159]: ill investigate and see if I can find better name

File: dot/core/messages.go (hunk 2 above)
Comment (jimjbrettj) [reply dot/core/messages.go:24]: I think this change should be fine

File: dot/core/messages.go (hunk 3 above)
Comment (jimjbrettj) [reply dot/core/messages.go:89]: I dont believe so. I think we want to still check for other valid transactions to add to the pool. We just need to know not to report the peer for good behavior if they have invalid txns i do believe

File: dot/core/messages_integration_test.go (hunk 7 above)
Comment (jimjbrettj) [reply dot/core/messages_integration_test.go:169]: its a boolean to indicate if the txns should be propagated. Renamed to shouldPropagate

File: dot/core/messages.go [hunk 11]
```diff
@@ -41,56 +78,29 @@ func (s *Service) HandleTransactionMessage(peerID peer.ID, msg *network.Transact
...
+		} else {
```
Comment (EclesioMeloJunior) [inline dot/core/messages.go:90]:
```suggestion
		} else  if validity.Propagate {
```

**Review by EclesioMeloJunior — CHANGES_REQUESTED** [review]

File: dot/core/messages.go (hunk 11 above)
Comment (jimjbrettj) [reply dot/core/messages.go]: @EclesioMeloJunior done!

**Review by EclesioMeloJunior — APPROVED** [review]

File: dot/core/messages.go [hunk 12]
```diff
@@ -19,9 +19,9 @@ import (
 func (s *Service) validateTransaction(peerID peer.ID, head *types.Header, rt runtime.Instance,
 	tx types.Extrinsic) (validity *transaction.Validity, valid bool, err error) {
 	s.storageState.Lock()
-	defer s.storageState.Unlock()
 
 	ts, err := s.storageState.TrieState(&head.StateRoot)
+	s.storageState.Unlock()
```
Comment (qdm12) [inline dot/core/messages.go:24]: Make sure it's fine to fiddle with `ts` with the storageState unlocked as well

**Pull request merged**